verifier/src/evidence.rs

```diff
@@ -165,14 +168,15 @@ quote_application_report_body_field_accessor! {
 /// This will perform most of the verification to be done on [`Evidence`] this includes:
 /// - verifying the certificate chains
 /// - verifying the QE identity (TODO)
-/// - verifying the TCB info (TODO)
+/// - verifying the TCB info
 /// - verifying the [`TrustedIdentity`] of the application enclave (TODO)
 /// - verifying the signature of the Quote (TODO)
 #[derive(Debug)]
 pub struct EvidenceVerifier<'a, C> {
     certificate_verifier: &'a C,
     _trusted_identities: Vec<TrustedIdentity>,
     time: DateTime,
+    tcb_verifier: Cell<Option<SignedTcbInfoVerifier>>,
```

Comment:
Not fond of the interior mutability with Cell here, but we can't build the `SignedTcbInfoVerifier` until we have the `Evidence`, since the key is in the evidence.

Perhaps this verifier should be different and it should take the evidence in the constructor as well as the verify method :shrug:

Reply:
Good idea!
I think I got too stuck on how it was done for other verifiers. Added the `SignedTcbInfoVerifier` to the `EvidenceValue`